packages/salesforcedx-apex-debugger/src/core/breakpointService.ts

```diff
@@ -219,7 +228,62 @@ export class BreakpointService {
     return Promise.resolve(this.getBreakpointsFor(uri));
   }
 
+  public async createExceptionBreakpoint(
+    projectPath: string,
+    sessionId: string,
+    typeref: string
+  ): Promise<string | undefined> {
+    const execution = new CliCommandExecutor(
+      new SfdxCommandBuilder()
+        .withArg('force:data:record:create')
+        .withFlag('--sobjecttype', 'ApexDebuggerBreakpoint')
+        .withFlag(
+          '--values',
+          `SessionId='${sessionId}' FileName='${typeref}' IsEnabled='true' Type='Exception'`
+        )
+        .withArg('--usetoolingapi')
+        .withArg('--json')
+        .build(),
+      { cwd: projectPath, env: RequestService.getEnvVars() }
+    ).execute();
+
+    const cmdOutput = new CommandOutput();
+    const result = await cmdOutput.getCmdResult(execution);
+    try {
+      const breakpointId = JSON.parse(result).result.id as string;
+      if (this.isApexDebuggerBreakpointId(breakpointId)) {
+        return Promise.resolve(breakpointId);
+      } else {
+        return Promise.reject(result);
+      }
+    } catch (e) {
+      return Promise.reject(result);
+    }
+  }
+
+  public async reconcileExceptionBreakpoints(
+    projectPath: string,
+    sessionId: string,
+    info: ExceptionBreakpointInfo
+  ): Promise<void> {
+    const knownBreakpointId = this.exceptionBreakpointCache.get(info.typeref);
+    if (knownBreakpointId && info.breakMode === 'never') {
```

Comment:
Breakpoint service will delete a breakpoint if it exists in the cache and the user choses `Never break` for the exception type.